### PR TITLE
Fix privilege requirement for CCS with Point In Time reader

### PR DESCRIPTION
**Pull request by ywangd**
When target indices are remote only, CCS does not require user to have privileges on the local cluster. This PR ensure Point-In-Time reader follows the same pattern.

Relates: #61827

@dnhatn @jimczi

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java [hunk 1]
```diff
@@ -479,6 +484,46 @@ public void testRemoteIndicesOnlyWorkWithApplicableRequestTypes() throws IOExcep
         verifyNoMoreInteractions(auditTrail);
     }
 
+    public void testUserWithNoRolesOpenPointInTimeWithRemoteIndices() {
+        final boolean hasLocalIndices = randomBoolean();
+        final String[] indices = new String[]{
+            hasLocalIndices ? randomAlphaOfLength(5) : "other_cluster:" + randomAlphaOfLength(5),
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java:490]: I would also exercise the condition that remote indices are wildcards.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java [hunk 2]
```diff
@@ -479,6 +484,46 @@ public void testRemoteIndicesOnlyWorkWithApplicableRequestTypes() throws IOExcep
         verifyNoMoreInteractions(auditTrail);
     }
 
+    public void testUserWithNoRolesOpenPointInTimeWithRemoteIndices() {
+        final boolean hasLocalIndices = randomBoolean();
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java:488]: set this to both values in a for-each.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java [hunk 3]
```diff
@@ -479,6 +484,46 @@ public void testRemoteIndicesOnlyWorkWithApplicableRequestTypes() throws IOExcep
...
+        verifyNoMoreInteractions(auditTrail);
+    }
+
+    public void testUserWithNoRolesCanClosePointInTime() {
+        final ClosePointInTimeRequest closePointInTimeRequest = new ClosePointInTimeRequest(randomAlphaOfLength(8));
+        final Authentication authentication = createAuthentication(new User("test user"));
+        mockEmptyMetadata();
+        final String requestId = AuditUtil.getOrGenerateRequestId(threadContext);
+        authorize(authentication, ClosePointInTimeAction.NAME, closePointInTimeRequest);
+        verify(auditTrail).accessGranted(eq(requestId), eq(authentication), eq(ClosePointInTimeAction.NAME), eq(closePointInTimeRequest),
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java:522]:
```suggestion
        verify(auditTrail).accessGranted(eq(requestId), eq(authentication), eq(indices:data/read/close_point_in_time), eq(closePointInTimeRequest),
```
If you wanna be consistent with the previous test.

File: x-pack/qa/multi-cluster-search-security/src/test/resources/rest-api-spec/test/multi_cluster/80_point_in_time.yml [hunk 4]
```diff
@@ -111,3 +111,39 @@ teardown:
...
+
+  - do:
+      headers: { Authorization: "Basic cmVtb3RlOnMza3JpdA==" }
+      open_point_in_time:
+        index: "my_*:point_in_time_index"
+        keep_alive: 5m
+  - set: {id: pit_id}
+
+  - do:
+      headers: { Authorization: "Basic am9lOnMza3JpdA==" }
```
Comment (albertzaharovits) [inline x-pack/qa/multi-cluster-search-security/src/test/resources/rest-api-spec/test/multi_cluster/80_point_in_time.yml:126]:
```suggestion
      headers: { Authorization: "Basic cmVtb3RlOnMza3JpdA==" }
```

Not only opening the PIT requires no privs on the local cluster but also using it (and closing), no?

**Review by albertzaharovits — APPROVED** [review]
LGTM Nice one Yang!

I had just a couple of small nits +
Also can you please amend the `IndicesAndAliasesResolverTests#testRemotableRequestsAllowRemoteIndices` test for the `OpenPointInTimeRequest` (and maybe randomly make `foo` a wildcard `foo*`).

**Comment by dnhatn** [comment]
@ywangd Thanks for fixing this.

**Comment by ywangd** [comment]
Thanks @albertzaharovits 

That was a good catch on the yaml test. You are right, neither subsequent search nor close remote PIT require any local privilege. It was a copy/paste error from the previous test's Authorization headers. I have now rectified them. Thanks!

**Pull request merged**